### ISSUE TITLE
480i doesn't work with forced_scandoubler on

### ISSUE THREAD

**Issue by alextirrell** [MiSTer-devel/Genesis_MiSTer#129]
I'm not sure how many titles there are that even switch between 240P and 480i, but Sonic 2's two player mode is one of them.

I'm using direct_video on a HDMI to VGA adapter while I wait for my IO Board.
My desired output device for my MiSTer is a CRT monitor, so I need 31khz and turn forced_scandoubler on.
Everything works perfect while the game is in 240P mode, but if you go into two player mode, it flips back to 15khz which is out of range. I've confirmed this is what it's doing--I brought it to my friend and we checked with his oscilloscope. 

I feel like I've exhausted every combination of options in the ini to account for it. 
The only thing that does work for 240P and 480i is turning off forced_scandoubler and direct_video and allowing the HDMI to VGA adapter to use the HDMI signal, which seems less than ideal. I already have no idea how it's going to work through the IO board.

**Comment by sorgelig** [comment]
Interlaced video cannot be scandoubled.

**Comment by mathieulh** [comment]
Indeed, you cannot scandouble an interlaced output without post processing, you would need something such as a bob deinterlace or Motion deinterlace feature, which is not currently implemented in MiSTer (I am not sure if @sorgelig or anyone else would want to spend time implementing this when, to my knowledge only 2 games supported on MiSTer (Sonic 2 and Sonic3) use 480i.

As a workaround you can have your output go through a GBS-C ( https://github.com/ramapcsx2/gbs-control/wiki/Build-the-Hardware ) or a similar upscaler and set 480p as passthrough, the GBS-C would leave the scandoubled output untouched while it would deinterlace 480i content. Of course if more games with interlaced output get supported in the future, implementing a bob deinterlace feature may become worthwhile. 

Perhaps there are other ways to handle this issue that I did not think of.

**Comment by alextirrell** [comment]
Actually, Sonic 3's 2 player mode doesn't use 480i--they had redesigned the 2P mode to just use entirely different sprites, so Sonic 2 really might be the only game on MiSTer right now that uses 480i? I haven't fully explored everything to find out--I just presumed there would be others and that's why I chose it to test. 

This would be a hurdle down the line with a PlayStation core though.

**Comment by boomlinde** [comment]
The original MiST manages to display interlaced modes on my VGA monitors just fine in e.g. Sonic 2 and Minimig-AGA while the scandoubler is enabled.

I'm not sure how it does this, and from memory I don't think it uses a frame buffer or scaler for this, but it looks quite good to me. It might be worth looking into. I don't really plan on playing the Sonic 2 multiplayer mode, but a lot of Amiga software uses interlaced modes, and for Amiga, with RTG and all, it does make sense to use a VGA monitor over a TV.

**Comment by HappehLemons** [comment]
Wonder if there's any workarounds for this, apart from Sonic 2s 480i, I can see PSX becoming an issue with this as well as it continues to develop. Using a VGA monitor with the scan doubled setting and I can't accept a 15khz frequency.

**Comment by boomlinde** [comment]
@HappehLemons The best workaround for now is probably to use the scaler and set `vga_scaler` to 1. This will redirect the scaler output to the analog output which should work for a VGA monitor.

This has a slight penalty of what I understand is normally a frame's worth of delay since the scaler needs to buffer an entire frame before it can apply scaling. This is the main reason I'd prefer not to use the scaler myself, but using the scaler should be better in terms of image quality: by buffering both fields you can properly deinterlace the picture (I'm not sure if the MiSTer scaler will do this, though).

There is another GitHub issue related to this, posed more as a feature request than a bug report: https://github.com/MiSTer-devel/Minimig-AGA_MiSTer/issues/130
